src/nupic/frameworks/opf/common_models/cluster_params.py

```diff
@@ -86,10 +90,14 @@ def getScalarMetricWithTimeOfDayAnomalyParams(metricData,
     maxVal = minVal + 1
 
   # Load model parameters and update encoder params
-  paramFileRelativePath = os.path.join(
-    "anomaly_params_random_encoder",
-    "best_single_metric_anomaly_params.json")
-
+  if (tmImplementation is "cpp"):
+    paramFileRelativePath = os.path.join(
+      "anomaly_params_random_encoder",
+      "best_single_metric_anomaly_params_cpp.json")
+  elif (tmImplementation is "tm_cpp"):
+    paramFileRelativePath = os.path.join(
+      "anomaly_params_random_encoder",
+      "best_single_metric_anomaly_params_tm_cpp.json")
```

Comment:
What happens if an incorrect value for `tmImplementation` is passed in? There should be an else that throws an appropriate exception.